File: ClaudeSessionReader.Server/Models/ClaudeModels.cs
namespace ClaudeSessionReader.Server.Models;

public record ClaudeProject(
    string FolderName,
    string FullPath,
    string DisplayName,
    string DecodedPath,
    int SessionCount,
    DateTime LastModified);

public record SubAgentInfo(
    string AgentId,
    string AgentType,
    string Description,
    int MessageCount);

public record ClaudeSession(
    string Id,
    string? Title,
    DateTime? StartTime,
    int MessageCount,
    List<SubAgentInfo> SubAgents);

public record ContentBlock(
    string Type,
    string? Text,
    string? Thinking,
    string? ToolUseId,
    string? ToolName,
    string? Input);

public record SessionMessage(
    string Uuid,
    string? ParentUuid,
    string Role,
    List<ContentBlock> Content,
    DateTime Timestamp);

public record ScanRequest(string Path);

public record BackupFileEntry(string RelativePath, long Size);

File: claudesessionreader.client/src/app/models/claude.models.ts
export interface ClaudeProject {
  folderName: string;
  fullPath: string;
  displayName: string;
  decodedPath: string;
  sessionCount: number;
  lastModified: string;
}

export interface SubAgentInfo {
  agentId: string;
  agentType: string;
  description: string;
  messageCount: number;
}

export interface ClaudeSession {
  id: string;
  title: string | null;
  startTime: string | null;
  messageCount: number;
  subAgents: SubAgentInfo[];
}

export interface ContentBlock {
  type: 'text' | 'thinking' | 'tool_use' | 'tool_result';
  text: string | null;
  thinking: string | null;
  toolUseId: string | null;
  toolName: string | null;
  input: string | null;
}

export interface SessionMessage {
  uuid: string;
  parentUuid: string | null;
  role: 'user' | 'assistant';
  content: ContentBlock[];
  timestamp: string;
}
